### DISCUSSION TITLE
Add support for `/.config`

### DISCUSSION THREAD

**Discussion by zoey-kaiser** [tailwindlabs/tailwindcss#12958] [Ideas]
Hello, everyone! 

Recently a new "movement" / "project" has started inside the Nuxt community. This was started by @pi0 and aims to motivate as many projects as possible to add support for adding their configuration files to a `./config` folder. You can read more about this [here](https://github.com/pi0/config-dir).

## Description

Due to a lack of standards for where tools store their configuration, the number of top-level configuration files in projects is increasing day by day. See [.config dir proposal](https://github.com/pi0/config-dir) for more context.

While it can take a while to onboard more tools with this new convention, it will be an egg-and-chicken problem to adopt a new convention like this unless more mainstream tools onboard to support it.

## Suggested solution

I propose supporting `.config/tailwindcss.js` and `.config/tailwindcss.config.js` as alternative places to check and load the configuration if the top-level `tailwindcss.config.js` does not exist to unblock adopting this new convention across the ecosystem.

I think this would need to mainly be done in https://github.com/tailwindlabs/tailwindcss-intellisense, and after doing some research I noticed that users can already define a custom location for this by using [`tailwindCSS.experimental.configFile`](https://github.com/tailwindlabs/tailwindcss-intellisense/tree/8d22df1b0b913c548e50a0548d807f861e9c5a2d?tab=readme-ov-file#tailwindcssexperimentalconfigfile), however it seems to be experimental currently. 

Based of this description in the repository:

> In order for the extension to activate you must have [tailwindcss installed](https://tailwindcss.com/docs/installation) and a [Tailwind config file](https://tailwindcss.com/docs/installation#create-your-configuration-file) named tailwind.config.{js,cjs,mjs,ts} in your workspace.

It seems that the intellisense can already find configs not inside the root, but I wanted to verify this and if it is not the case, request that `./config/tailwind.config.js` and `./config/tailwind.js` could also be added to the default locations to search for!

## Alternatives

See [relevant discussions](https://github.com/pi0/config-dir/discussions).

## Context

We approached the Nuxt Module of TailwindCSS to add support for this, however we also wanted to bring this into the main project to see if we can motivate more developers outside of the Nuxt sphere to also adopt this!

https://github.com/nuxt-modules/tailwindcss/issues/808

**Comment by ineshbose** [comment]
I'll link the PR that added supported for ESM/TS configuration files - https://github.com/tailwindlabs/tailwindcss/pull/10785

This loader already uses `jiti`, so usage of [`c12`](https://github.com/unjs/c12) is much better, but I'm aware (I read some discussion on an issue) that this custom loader had to be used to keep the synchronous code across the repository, otherwise a lot of functions would be updated to be `async`.

**Reply by pi0** [reply]
Thanks for the ref. Yes probably using all c12 just for `.config` dir would be overkill here. (although in next major version of jiti, we will bring opt-in `jiti.import` util with better ESM support. probably another story for another day haha)

What here we need is something as simple as a try-catch for search path to check `.config/tailwind` as alternative (like [this](https://github.com/unjs/c12/blob/4d321bac5374100cb249b5bd6afa5c16f3488270/src/loader.ts#L331))
